src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -378,26 +391,94 @@ class MediaBaseViewer extends BaseViewer {
     }
 
     /**
-     * Toggle playback
+     * Removes pause event listener
+     *
+     * @private
+     * @return {void}
+     */
+    removePauseEventListener() {
+        this.mediaEl.removeEventListener('timeupdate', this.pauseListener);
+    }
+
+    /**
+     * Validates time parameter
      *
      * @private
+     * @param {number} time - time for media
+     * @return {boolean} - true if time is valid
+     */
+    isValidTime(time) {
+        return typeof time === 'number' && isFinite(time) && time >= 0 && time <= this.mediaEl.duration;
+    }
+
+    /**
+     * Play media, optionally from start time to end time
+     * @param {number} start - start time in seconds
+     * @param {number} end - end time in seconds
      * @emits play
-     * @emits pause
      * @return {void}
      */
-    togglePlay() {
-        if (this.mediaEl.paused) {
+    play(start, end) {
+        const hasValidStart = this.isValidTime(start);
+        const hasValidEnd = this.isValidTime(end);
+        this.removePauseEventListener();
+        if (hasValidStart) {
+            if (hasValidEnd) {
+                if (start < end) {
+                    this.pause(end);
+                }
+            }
+            // Start playing media from <start> time
+            this.setMediaTime(start);
+        }
+        if (arguments.length === 0 || hasValidStart) {
             this.mediaEl.play();
             this.emit('play');
             this.handleRate();
             this.handleVolume();
+        }
+    }
+
+    /**
+     * Pause media
+     * @param {number} time - time at which media is paused
+     * @emits pause
+     * @return {void}
+     */
+    pause(time) {
+        const hasValidTime = this.isValidTime(time);
+        // Remove eventListener because segment completed playing or user paused manually
+        this.removePauseEventListener();
+        if (hasValidTime) {
+            this.pauseListener = () => {
+                if (this.mediaEl.currentTime > time) {
+                    this.pause();
```

Comment:
emit pause here too

Reply:
not required here because when `this.pause()` will `emit('pause')`.